server/src/test/java/org/opensearch/common/util/BatchRunnableExecutorTests.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.util;
+
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.common.util.concurrent.TimeoutAwareRunnable;
+import org.opensearch.test.OpenSearchTestCase;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Supplier;
+
+import org.mockito.ArgumentCaptor;
+
+import static org.mockito.Mockito.atMost;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class BatchRunnableExecutorTests extends OpenSearchTestCase {
+    private Supplier<TimeValue> timeoutSupplier;
+    private TimeoutAwareRunnable runnable1;
+    private TimeoutAwareRunnable runnable2;
+    private TimeoutAwareRunnable runnable3;
+    private List<TimeoutAwareRunnable> runnableList;
+
+    public void setupRunnables() {
+        timeoutSupplier = mock(Supplier.class);
+        runnable1 = mock(TimeoutAwareRunnable.class);
+        runnable2 = mock(TimeoutAwareRunnable.class);
+        runnable3 = mock(TimeoutAwareRunnable.class);
+        runnableList = Arrays.asList(runnable1, runnable2, runnable3);
+    }
+
+    public void testRunWithoutTimeout() {
+        setupRunnables();
+        when(timeoutSupplier.get()).thenReturn(TimeValue.timeValueSeconds(1));
```

Comment:
I think the following works and is simpler:

```suggestion
        timeoutSupplier = () -> TimeValue.timeValueSeconds(1);
```

Reply:
Updated